pkg/helm/charts.go

```diff
@@ -0,0 +1,75 @@
+// Copyright 2020 The Lokomotive Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package helm handles Helm-related operations.
+package helm
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"github.com/kinvolk/lokomotive/pkg/assets"
+	"github.com/pkg/errors"
+	"helm.sh/helm/v3/pkg/chart"
+	"helm.sh/helm/v3/pkg/chart/loader"
+)
+
+// ComponentChart finds a chart for a component named name in the assets and returns a pointer to a
+// chart.Chart struct.
+func ComponentChart(name string) (*chart.Chart, error) {
+	p := filepath.Join(assets.ComponentsSource, name)
+
+	c, err := chartFromAssets(p)
+	if err != nil {
+		return nil, fmt.Errorf("loading chart from assets: %w", err)
+	}
+
+	return c, nil
+}
+
+// ControlPlaneChart finds a chart for a control plane component named name in the assets and
+// returns a pointer to a chart.Chart struct.
+func ControlPlaneChart(name string) (*chart.Chart, error) {
+	p := filepath.Join(assets.ControlPlaneSource, name)
+
+	c, err := chartFromAssets(p)
+	if err != nil {
+		return nil, fmt.Errorf("loading chart from assets: %w", err)
+	}
+
+	return c, nil
+}
+
+// chartFromAssets finds a Helm chart in the assets at location, creates a chart.Chart struct from
+// the chart and returns a pointer to it.
+func chartFromAssets(location string) (*chart.Chart, error) {
```

Comment:
IMO this function should be exported and then `pkg/platform` (as we don't have a better place for it right now) and `pkg/components` should export their own respective `Chart()` functions. This way, both names and dependencies will compose more nicely.